cmake/modules/RootNewMacros.cmake

```diff
@@ -603,7 +604,9 @@ function(ROOT_LINKER_LIBRARY library)
     # ROOT_GENERATE_DICTIONARY call that prevents this warning.
     #message(AUTHOR_WARNING "Couldn't find target: " ${library} "\n Forgot to call ROOT_GENERATE_DICTIONARY?")
   endif()
-  add_dependencies(${library} move_headers)
+  if(NOT ARG_NOMOVEHEADERSDEP)
```

Comment:
I think we could also check for `CMAKE_PROJECT_NAME` equals `ROOT` which means that users don't have to figure out they have to call it with this parameter. Or at least we should rename the parameter something more generic like "ROOT_EXTERNAL" because we might add more special cases for external projects here. @peremato what do you think?

Reply:
This is right. Better to check for `CMAKE_PROJECT_NAME` to be or not to be `ROOT`.

Reply:
Updated as requested.